mne/decoding/tests/test_search_light.py

```diff
@@ -236,3 +236,36 @@ def test_generalization_light():
         features_shape = pipe.estimators_[0].steps[0][1].features_shape_
         assert_array_equal(features_shape, [3, 4])
     assert_array_equal(y_preds[0], y_preds[1])
+
+
+@requires_version('sklearn', '0.17')
+def test_cross_val_predict():
+    """Test cross_val_predict with predict_proba."""
+    from sklearn.linear_model import LogisticRegression, LinearRegression
+    from sklearn.base import BaseEstimator, clone
+    from sklearn.model_selection import cross_val_predict
+    X = np.random.randn(10, 5, 3)
+    y = np.random.randint(0, 2, 10)
+
+    estimator = SlidingEstimator(LinearRegression())
+    cross_val_predict(estimator, X, y, cv=2)
+
+    class Classifier(BaseEstimator):
+        """Moch class that does not have classes_ attribute"""
+        def __init__(self):
+            self.base_estimator = LogisticRegression()
+
+        def fit(self, X, y):
+            self.estimator_ = clone(self.base_estimator).fit(X, y)
+            return self
+
+        def predict_proba(self, X):
+            return self.estimator_.predict_proba(X)
+
+    with warnings.catch_warnings(record=True):
```

Comment:
Can you add a comment which warnings you are catching here? It's nice to have so that eventually we can remove these where possible.